tests/contrib/oed/test_eig.py

```diff
@@ -1,57 +1,241 @@
+        "w",
+        donsker_varadhan_eig,
+        [200, 800, GuideDV(basic_2p_ba_guide(5)),
+         optim.Adam({"lr": 0.025}), False, None, 500],
+        True,
+        0.3
+    ),
+    T(
+        basic_2p_linear_model_sds_10_2pt5,
+        AB_test_2d_10n_2p,
+        "y",
+        "w",
+        barber_agakov_ape,
+        [20, 400, basic_2p_ba_guide(2), optim.Adam({"lr": 0.05}),
+         False, None, 500],
+        False,
+        0.2
+    ),
+    T(
+        basic_2p_linear_model_sds_10_2pt5,
+        X_circle_5d_1n_2p,
+        "y",
+        "w",
+        barber_agakov_ape,
+        [20, 400, basic_2p_ba_guide(5), optim.Adam({"lr": 0.05}),
+         False, None, 500],
+        False,
+        0.2
+    ),
+    T(
+        group_2p_linear_model_sds_10_2pt5,
+        X_circle_5d_1n_2p,
+        "y",
+        "w1",
+        naive_rainforth_eig,
+        [400, 400, 400],
+        True,
+        0.2
+    ),
+    # This fails because guide is wrong
+    pytest.param(
+        group_2p_linear_model_sds_10_2pt5,
+        X_circle_5d_1n_2p,
+        "y",
+        "w1",
+        vi_ape,
+        [{"guide": group_2p_guide, "optim": optim.Adam({"lr": 0.05}),
+          "loss": elbo, "num_steps": 1000}, {"num_samples": 1}],
+        False,
+        0.3,
+        marks=pytest.mark.xfail
+    ),
+    T(
+        group_2p_linear_model_sds_10_2pt5,
+        X_circle_5d_1n_2p,
+        "y",
+        "w1",
+        donsker_varadhan_eig,
+        [200, 400, GuideDV(group_2p_ba_guide(5)),
+         optim.Adam({"lr": 0.025}), False, None, 500],
+        True,
+        0.3
+    ),
+    T(
+        group_2p_linear_model_sds_10_2pt5,
+        X_circle_5d_1n_2p,
+        "y",
+        "w1",
+        barber_agakov_ape,
+        [20, 400, group_2p_ba_guide(5), optim.Adam({"lr": 0.05}),
+         False, None, 500],
+        False,
+        0.2
+    )
+]
+
+
+@pytest.mark.parametrize("model,design,observation_labels,target_labels,estimator,args,eig,allow_error", TEST_CASES)
+def test_eig_lm(model, design, observation_labels, target_labels, estimator, args, eig, allow_error):
     pyro.set_rng_seed(42)
     pyro.clear_param_store()
-    vi_parameters = {
-        "guide": lambda d: guide(arg, d),
-        # Note: exact details irrelevant, taking 0 steps
-        "optim": optim.Adam({"lr": 0.01}),
-        "loss": Trace_ELBO(),
-        "num_steps": n_steps
-    }
-    is_parameters = {"num_samples": 1}
-    ape = vi_ape(lambda d: model(arg, d), design, "y", ["w", "u"], vi_parameters,
-                 is_parameters)
-    assert_equal(ape, expected_ape, prec=1e-4)
+    y = estimator(model, design, observation_labels, target_labels, *args)
+    if model is bernoulli_model:
+        y_true = bernoulli_ground_truth(model, design, observation_labels, target_labels, eig=eig)
+    else:
+        y_true = linear_model_ground_truth(model, design, observation_labels, target_labels, eig=eig)
+    print()
+    print(estimator.__name__)
+    print(y)
+    print(y_true)
```

Comment:
are these debugging statements? why do you need to print these since theyll be caught in the assert? you should use a logger if so